src/networkWrapper/wrappers/socketio.js

```diff
@@ -12,11 +22,75 @@ function SocketIO(host, port, ssl) {
    * @param {int} reconnectionDelay
    */
   this.connect = function (autoReconnect, reconnectionDelay) {
+    var self = this;
+
     this.socket = window.io((this.ssl ? 'https://' : 'http://') + this.host + ':' + this.port, {
       reconnection: autoReconnect,
       reconnectionDelay: reconnectionDelay,
       forceNew: true
     });
+
+    this.socket.on('connect', function() {
+      if (self.wasConnected) {
```

Comment:
According to socket.io documentation, `connect` event is fired "upon a connection including a successful reconnection".
If this is the case, don't we have a double trigger with the `on('reconnect',..)` one?